### PR TITLE
QTY-947

### PR DESCRIPTION
**Pull request by JamesWingert**
[QTY-947](https://strongmind.atlassian.net/browse/QTY-947)

## Purpose 
- Add Launch Darkly to Canvas
- Upgrade Ruby 2.5.0 -> 2.6.10

## Approach 
- Add and test a feature flag from Launch Darkly UI and see if test environment reflects the change.
- Ensure Ruby upgrade is successful.
- Once flag is reflecting accurately in test environment, change the codebase to get the current user session so the flag can be applied to specific user base.
- Verify user is appearing in Users list in Launch Darkly UI

## Testing
- Tested locally with hard coded user information
- Tested locally with user session
- Verified Launch Darkly UI changes reflected accurately for both cases

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2262,4 +2262,16 @@ def setup_live_events_context
   def teardown_live_events_context
     LiveEvents.clear_context!
   end
+
+  def launch_darkly_user
+    domain = ENV['SETTINGS_TABLE_PREFIX'] || 'localhost'
```
Comment (CJAdeszko) [inline app/controllers/application_controller.rb:2267]: is `|| 'localhost'` necessary here? I think the `SETTINGS_TABLE_PREFIX` was only set to `'settings'` on an out of date version of the `canvas.env` file? I may be wrong on that though...

File: app/views/users/dashboard_sidebar.html.erb [hunk 2]
```diff
@@ -41,10 +41,12 @@
 <% end %>
 <div>
   <% if show_user_create_course_button(@current_user) %>
+    <% if @show_feature %>
```
Comment (CJAdeszko) [inline app/views/users/dashboard_sidebar.html.erb:44]: Are we intentionally leaving this in or was this only supposed to be a test of the gem functionality?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (CJAdeszko) [reply app/controllers/application_controller.rb:2267]: Also, why are we adding this to the `ApplicationController` in LMS and the shim?

File: app/views/users/dashboard_sidebar.html.erb (hunk 2 above)
Comment (adiaz04) [reply app/views/users/dashboard_sidebar.html.erb:44]: should be removed.

File: app/views/users/user_dashboard.html.erb [hunk 3]
```diff
@@ -40,6 +40,8 @@
   <div id="dashboard_header_container" class="ic-Dashboard-header">
     <h1 class="ic-Dashboard-header__title"><%= t "Dashboard" %></h1>
   </div>
+
+  </div>
```
Comment (adiaz04) [inline app/views/users/user_dashboard.html.erb:44]: remove extra `</div>`

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (adiaz04) [reply app/controllers/application_controller.rb:2267]: shim shouldn't have any changes for this feature. will remove.

Comment (adiaz04) [reply app/controllers/application_controller.rb:2267]: yea - it looks like the SETTINGS_TABLE_PREFIX is required so we don't need to have the default localhost here.

**Review by CJAdeszko — APPROVED** [review]
👍

**Review by agtorre3 — APPROVED** [review]

**Pull request merged**